tools/Bridge/Bridge/IdleTimeoutManager.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Web.Http;
+
+namespace Bridge
+{
+    public class IdleTimeoutHandler : DelegatingHandler
+    {
+        IdleTimeoutManager _timeoutManager;
+
+        private IdleTimeoutHandler(TimeSpan idleTimeout)
+        {
+            _timeoutManager = new IdleTimeoutManager(idleTimeout); ;
+            _timeoutManager.OnTimeOut += (s, e) =>
+            {
+                Console.WriteLine("Timed out as there were no messages to the bridge for {0} seconds", (int)e.TotalSeconds);
+                Environment.Exit(-1);
```

Comment:
NIT: Consider defining exit codes as an enum or some similar mechanism, e.g. `Success = 0, TimedOut = -1`, so we're not confused about why we're exiting. Also future-proofing in case we start having other exit codes.